framework/src/constraints/MortarConstraintBase.C

```diff
@@ -68,6 +68,12 @@ MortarConstraintBase::validParams()
       "compute_primal_residuals", true, "Whether to compute residuals for the primal variable.");
   params.addParam<bool>(
       "compute_lm_residuals", true, "Whether to compute Lagrange Multiplier residuals");
+  params.addParam<bool>(
+      "interpolate_normals",
+      true,
+      "Whether to interpolate the nodal normals (e.g. classic idea of evaluating field at "
+      "quadrature points). If this is set to false, then non-interpolated nodal normals will be "
```

Comment:
The non-interpolated normal normals are still averaged compared to the element face normals